android/fretboard.md

```diff
@@ -0,0 +1,141 @@
+You will need to choose an experiment source. Fretboard currently supports two choices for experiment sources. It can either read from a [Kinto](https://github.com/Kinto/kinto) server or from a JSON file on local disk. Using Kinto has the benefit of offering buckets, so you can split your users into an A and a B population.
+
+## Synchronizing Experiments from the Source
+There are two calls used for synchronizing experiments. One, Fretboard.loadExperiments() which reads current experiment values from a file on disk and a second call, Fretboard.updateExperiments() that goes out to the ExperimentSource. Both calls should not be made from Android main/UI thread. Any one of the multithreading methods on Android should be used to perform the work asynchronously from a background thread: Kotlin Coroutines, RxJava, AsyncTasks, Executors, etc.
+
+You may want to call loadExperiments() before calling updateExperiments() so archived values from past runs are loaded quickly from disk before waiting for experiment values to be updated from the network.
+
+You can learn about these calls from the [Fretboard documentation](https://github.com/mozilla-mobile/android-components/tree/master/components/service/fretboard#fetching-experiments-from-disk).
+
+## Experiment Filters
+In addition to separating users into buckets, you may decide to limit your experiment to a user population with specific characteristics, such as a particular app version, app ID, country, language, or device manufacturer. This is done using experiment filters, which is described in the [Fretboard documentation](https://github.com/mozilla-mobile/android-components/tree/master/components/service/fretboard#filters)
+
+You may even create your own filters by defining a custom ValuesProvider.
+
+## Experiment Overrides
+Fretboard Overrides offer a means of locally changing which experiments are enabled for the current device. You can set an override and the server configuration will no longer choose which experiments are set. These values are stored locally in Android shared properties, so they will be wiped when a user clears app data.
+
+There are methods to both set and clear overrides using either blocking or non-blocking calls. If you clear an override, then that device will resume receiving experiments from the ExperimentSource. Using a blocking call will ensure that the override is set before the next statement gets executed, but it is forbidden and will cause an error to use blocking calls from Android's main/UI thread.
+
+Learn more about overrides in the [Fretboard documentation](https://github.com/mozilla-mobile/android-components/tree/master/components/service/fretboard#setting-override-values).
+
+## Experiment Schema For Kinto
+Below is an example JSON schema for Kinto, which can be used to provide the default filters for Fretboard experiments. You will have to change this slightly if you define your own filters or use more buckets than 0 through 100.
+
+`{
```

Comment:
This came out unformatted:
![screen shot 2018-09-14 at 11 15 00](https://user-images.githubusercontent.com/759372/45567742-7a428480-b80f-11e8-844a-4f86abd69462.png)

Did you mean <pre>```json</pre> then a newline and then the JSON?

Reply:
Fixed